Components/CategoryResolver/DefaultCategoryResolver.php

```diff
@@ -103,7 +115,31 @@ public function storeRemoteCategories(array $categories, $articleId)
             $productToCategory->setConnectCategory($remoteCategory);
             $this->manager->persist($productToCategory);
         }
+    }
 
-        $this->manager->flush();
+    /**
+     * @param \Shopware\CustomModels\Connect\RemoteCategory[] $assignedCategories
+     * @param $articleId
+     */
+    private function removeProductsFromRemoteCategory($assignedCategories, $articleId)
+    {
+        $currentProductCategories = $this->productToRemoteCategoryRepository->getArticleRemoteCategories($articleId);
+
+        /** @var ProductToRemoteCategory $currentProductCategory */
+        foreach ($currentProductCategories as $currentProductCategory) {
```

Comment:
@sbofirov sure that this one will work?

if (!in_array($currentProductCategory, $assignedCategoryIds) {
$this->manager->remove($currentProductCategory);
}

Because the Array is an Array of Objects... and the assignedCategoryIds are integer values... is this working? I never tried....

Reply:
Ahh yes, good point. must be ```$currentProductCategory->getConnectCategoryId()``` instead of ```$currentProductCategory```